iac/notary/modules/deploy/main.tf

```diff
@@ -0,0 +1,246 @@
+  query = {
+    registry = "${azurerm_container_registry.acr.name}"
+    tokenName = "exampletoken"
+  }
+  depends_on = [
+    azurerm_container_registry.acr
+  ]
+}
+
+resource "azurerm_key_vault_secret" "notationUser" {
+  name         = "NOTATION-USERNAME"
+  value        = data.external.token.result["name"]
+  key_vault_id = azurerm_key_vault.kv.id
+  depends_on = [
+    data.external.token,
+    azurerm_key_vault_access_policy.currentUserAccesPolicy
+  ]
+}
+
+resource "azurerm_key_vault_secret" "notationPassword" {
+  name         = "NOTATION-PASSWORD"
+  value        = data.external.token.result["password"]
+  key_vault_id = azurerm_key_vault.kv.id
+  depends_on = [
+    data.external.token,
+    azurerm_key_vault_access_policy.currentUserAccesPolicy
+  ]
+}
+
+resource "azurerm_kubernetes_cluster" "aks" {
+  name                             = "${random_id.main.hex}-aks"
+  location                         = azurerm_resource_group.example.location
+  resource_group_name              = azurerm_resource_group.example.name
+  dns_prefix                       = "${random_id.main.hex}-aks-dns"
+  kubernetes_version               = "1.22.11"
+  sku_tier                         = "Free"
+  http_application_routing_enabled = true
+
+
+  network_profile {
+    network_plugin    = "kubenet"
+    load_balancer_sku = "standard"
+  }
+
+
+  azure_active_directory_role_based_access_control {
+    managed            = true
+    azure_rbac_enabled = true
+  }
+
+  default_node_pool {
+    name            = "default"
+    node_count      = 1
+    max_pods        = 110
+    vm_size         = "Standard_D2_v2"
+    os_disk_size_gb = 128
+    os_sku          = "Ubuntu"
+    zones           = ["1"]
+  }
+
+  service_principal {
+    client_id     = azuread_service_principal.example.application_id
+    client_secret = azuread_service_principal_password.example.value
+  }
+  depends_on = [
+    azuread_service_principal.example
+  ]
+}
+
+resource "azurerm_role_assignment" "linkAcrAks" {
+  principal_id                     = azuread_service_principal.example.object_id
+  role_definition_name             = "AcrPull"
+  scope                            = azurerm_container_registry.acr.id
+  skip_service_principal_aad_check = true
+}
+
+resource "azurerm_key_vault_certificate" "signingCert" {
+  name         = "example"
+  key_vault_id = azurerm_key_vault.kv.id
+
+  certificate_policy {
+    issuer_parameters {
+      name = "Self"
+    }
+
+    key_properties {
+      exportable = true
+      key_size   = 2048
+      key_type   = "RSA"
+      reuse_key  = true
+    }
+
+    lifetime_action {
+      action {
+        action_type = "AutoRenew"
+      }
+
+      trigger {
+        days_before_expiry = 30
+      }
+    }
+
+    secret_properties {
+      content_type = "application/x-pkcs12"
+    }
+
+    x509_certificate_properties {
+      extended_key_usage = ["1.3.6.1.5.5.7.3.1","1.3.6.1.5.5.7.3.2","1.3.6.1.5.5.7.3.3"]
```

Comment:
Please only use the codesign EKU.  This is updated in the public azure doc.

Reply:
Just to confirm the codesign EKU is `1.3.6.1.5.5.7.3.3` correct? I couldn't find any docs specific to Azure, but I did find this https://ldapwiki.com/wiki/ExtendedKeyUsage.

Reply:
Correct.  Referenced here: https://learn.microsoft.com/en-us/azure/container-registry/container-registry-tutorial-sign-build-push#create-a-self-signed-certificate-azure-cli

Reply:
Thanks! I've updated the code to only use the code sign EKU.